test/conftest.py

```diff
@@ -90,16 +102,16 @@ def ipv6_san_server(tmp_path_factory):
 
 
 @pytest.fixture
-def stub_timezone(request):
+def stub_timezone(request: pytest.FixtureRequest) -> Generator[None, None, None]:
     """
     A pytest fixture that runs the test with a stub timezone.
     """
-    with stub_timezone_ctx(request.param):
+    with stub_timezone_ctx(request.param):  # type: ignore[attr-defined]
         yield
 
 
 @pytest.fixture(scope="session")
-def supported_tls_versions():
+def supported_tls_versions() -> Set[Optional[str]]:
```

Comment:
Given this is a shared session fixture, it will be safer to use `AbstractSet` instead of `Set`, to ensure no user mutates it, affecting all other users.

Also, why is the `Optional` needed? (can `_sock.version()` be None?)

Reply:
> Given this is a shared session fixture, it will be safer to use AbstractSet instead of Set, to ensure no user mutates it, affecting all other users.

Good to know. Thanks!

> why is the Optional needed? (can _sock.version() be None?)

Yes, `_sock.version()` returns `Optional[str])`